### PR TITLE
ODE-1056 Add SDW Depositor Submodule to ODE

### PR DESCRIPTION
**Pull request by mvs5465**
### Overview

This adds the [SDW depositor](https://github.com/usdot-jpo-ode/jpo-sdw-depositor) as a submodule to the ODE as part of ODE-944, ODE-1056, and ODE-1098.

### Tasks

- [x] Add SDW depositor as git submodule
- [x] Add SDW depositor to docker-compose files
- [x] Change from DDS websocket deposit to SDW kafka topic publish in ODE code
- [x] Unit test ODE code
- [x] User guide documentation
- [x] README documentation
- [x] Wiki documentation

### CODE REVIEW

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1CommandManager.java [hunk 1]
```diff
@@ -75,17 +73,22 @@ public Asn1CommandManagerException(String msg, Exception e) {
...
 
       this.odeProperties = odeProperties;
 
       this.signatureUri = odeProperties.getSecuritySvcsSignatureUri();
 
       try {
-         depositor = new DdsDepositor<>(odeProperties);
+         this.stringMessageProducer = MessageProducer.defaultStringMessageProducer(odeProperties.getKafkaBrokers(),
+               odeProperties.getKafkaProducerType(), odeProperties.getKafkaTopicsDisabledSet());
+         this.setDepositTopic(odeProperties.getKafkaTopicSdwDepositorInput());
```
Comment (hmusavi) [inline jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1CommandManager.java]: Let's leave the WebSocket interface still functional and switchable.

File: README.md [hunk 2]
```diff
@@ -201,6 +202,7 @@ git clone --recurse-submodules https://github.com/usdot-jpo-ode/jpo-ode.git
 - S3 Bucket Depositor - [jpo-s3-deposit](https://github.com/usdot-jpo-ode/jpo-s3-deposit)
 - Security Services Module- [jpo-security](https://github.com/usdot-jpo-ode/jpo-security-svcs)
 - ASN.1 CODEC - [asn1_codec](https://github.com/usdot-jpo-ode/asn1_codec)
+- SDW Bucket Depositor - [jpo-sdw-depositor](https://github.com/usdot-jpo-ode/jpo-sdw-depositor)
```
Comment (hmusavi) [inline README.md]: SDW Bucket? :)

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1CommandManager.java [hunk 3]
```diff
@@ -94,14 +102,19 @@ public Asn1CommandManager(OdeProperties odeProperties) {
...
+            depositor.deposit(asdBytes);
+            logger.info("Deposited message to SDW directly via websocket {}", asdBytes);
+         } catch (DdsRequestManagerException e) {
+            String msg = "Failed to deposit message to SDW";
+            throw new Asn1CommandManagerException(msg, e);
+         }
+      } else {
+         stringMessageProducer.send(this.getDepositTopic(), null, asdBytes);
+         logger.info("Published message to SDW deposit Kafka topic {}", asdBytes);
       }
```
Comment (hmusavi) [inline jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1CommandManager.java]: Missing EventLoger: `EventLogger.logger.info("Message Deposited to SDW: {}", asdBytes);`

**Comment by mvs5465** [comment]
https://github.com/usdot-jpo-ode/jpo-ode/wiki/Migrating-from-SDW-websocket-depositor-to-SDW-Depositor-Submodule Added transition wiki page for instructions on how to migrate from the existing DDS deposit to the new SDW Depositor Submodule

File: docker-compose-ppm-nsv.yml [hunk 4]
```diff
@@ -49,15 +49,17 @@ services:
...
       ODE_SECURITY_SVCS_SIGNATURE_URI: ${ODE_SECURITY_SVCS_SIGNATURE_URI}
       ODE_RSU_USERNAME: ${ODE_RSU_USERNAME}
       ODE_RSU_PASSWORD: ${ODE_RSU_PASSWORD}
-# Commented out for latest schemaVersion. Uncomment to set for older schemaVersion
-#      ODE_OUTPUT_SCHEMA_VERSION: ${ODE_OUTPUT_SCHEMA_VERSION}
+      ODE_DDS_CAS_USERNAME: ${ODE_DDS_CAS_USERNAME}
+      ODE_DDS_CAS_PASSWORD: ${ODE_DDS_CAS_PASSWORD}
+      # Commented out, will use SDW depositor module by default
+      #        ODE_DEPOSIT_SDW_MESSAGES_OVER_WEBSOCKET: ${ODE_DEPOSIT_SDW_MESSAGES_OVER_WEBSOCKET}
+      # Commented out for latest schemaVersion. Uncomment to set for older schemaVersion
```
Comment (hmusavi) [inline docker-compose-ppm-nsv.yml]: We don't need or use `ODE_OUTPUT_SCHEMA_VERSION` any more. We can remove it.

File: docker-compose.yml [hunk 5]
```diff
@@ -44,13 +44,15 @@ services:
...
       ODE_SECURITY_SVCS_SIGNATURE_URI: ${ODE_SECURITY_SVCS_SIGNATURE_URI}
       ODE_RSU_USERNAME: ${ODE_RSU_USERNAME}
       ODE_RSU_PASSWORD: ${ODE_RSU_PASSWORD}
-# Commented out for latest schemaVersion. Uncomment to set for older schemaVersion
-#      ODE_OUTPUT_SCHEMA_VERSION: ${ODE_OUTPUT_SCHEMA_VERSION}
+      ODE_DDS_CAS_USERNAME: ${ODE_DDS_CAS_USERNAME}
+      ODE_DDS_CAS_PASSWORD: ${ODE_DDS_CAS_PASSWORD}
+      # Commented out, will use SDW depositor module by default
+      #        ODE_DEPOSIT_SDW_MESSAGES_OVER_WEBSOCKET: ${ODE_DEPOSIT_SDW_MESSAGES_OVER_WEBSOCKET}
+      # Commented out for latest schemaVersion. Uncomment to set for older schemaVersion
```
Comment (hmusavi) [inline docker-compose.yml]: We don't use `ODE_OUTPUT_SCHEMA_VERSION` any more. We can remove it.

Comment (mvs5465) [reply docker-compose.yml]: Done

**Pull request merged**